pkg/controller/gitopsservice/gitopsservice_controller.go

```diff
@@ -213,6 +241,60 @@ func (r *ReconcileGitopsService) Reconcile(request reconcile.Request) (reconcile
 		}
 	}
 
+	// Define Service account for backend Service
+	serviceAccountObj := newServiceAccount(serviceNamespacedName)
+
+	// Set GitopsService instance as the owner and controller
+	if err := controllerutil.SetControllerReference(instance, serviceAccountObj, r.scheme); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	existingServiceAccount := &corev1.ServiceAccount{}
+	err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: serviceAccountObj.Namespace, Name: serviceAccountObj.Name}, existingServiceAccount)
```

Comment:
@shubhamagarwal19  Before this PR, which service account did this run as ? Why can't we create [cluster]rolebindings on that one?

Reply:
it was using the `default` serviceAccount which doesn't have any permissions ... so it now creates a ClusterRole and RoleBinding which lets it `GET/LIST/WATCH` `Application resource in argoproj.io api` for all namespaces.

Reply:
Thanks, any reason we shouldn't add the permissions to the `default` sa ?

Reply:
I think `default` SA is created by the k8s itself for each namespace. So I don't think it would be a nice idea to alter that? Also, any resource hor which `serivceAccount` isn't specified explicitly, uses the `default` SA for that namespace. 
So it would then have the permissions it shouldn't have.

Reply:
> Also, any resource where serivceAccount isn't specified explicitly, uses the default SA for that namespace, so it would then have the permissions it shouldn't have.

Makes sense, thank you 👍 

Have we done due diligence to ensure smooth upgrades?